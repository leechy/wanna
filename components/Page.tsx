import { View } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';

interface PageProps {
  children: React.ReactNode;
}

export default function Page({ children }: PageProps) {
  const { top } = useSafeAreaInsets();

  return <View style={{ marginTop: top, flex: 1 }}>{children}</View>;
}
