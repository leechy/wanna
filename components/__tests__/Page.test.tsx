// FILE: components/Page.test.tsx

import React from 'react';
import { render } from '@testing-library/react-native';

import Page from '@/components/Page';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { View } from 'react-native';

jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: jest.fn(),
}));

describe('Page component', () => {
  it('matches snapshot', () => {
    (useSafeAreaInsets as jest.Mock).mockReturnValue({ top: 0 });

    const { toJSON } = render(
      <Page>
        <View />
      </Page>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with top inset', () => {
    (useSafeAreaInsets as jest.Mock).mockReturnValue({ top: 10 });

    const { getByTestId } = render(
      <Page>
        <View testID="child-view" />
      </Page>
    );

    const childView = getByTestId('child-view');
    expect(childView).toBeTruthy();
  });

  it('applies the correct marginTop style based on top inset', () => {
    (useSafeAreaInsets as jest.Mock).mockReturnValue({ top: 20 });

    const { getByTestId } = render(
      <Page>
        <View testID="child-view" />
      </Page>
    );

    const pageView = getByTestId('child-view').parent.parent;
    expect(pageView.props.style).toMatchObject({ marginTop: 20, flex: 1 });
  });
});
