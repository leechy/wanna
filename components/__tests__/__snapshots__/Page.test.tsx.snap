// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component matches snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "marginTop": 0,
      "paddingTop": 18,
    }
  }
>
  <View />
</View>
`;
